pkg/arvo/sys/hoon.hoon

```diff
@@ -1897,6 +1898,1491 @@
     ?~  a  ~
     ?~(r.a [~ n.a] $(a r.a))
   --
+::                                                      ::
+::::  2r: psq logic                                     ::
+  ::                                                    ::
+  ::
+++  up
+  ~%  %up  ..up  ~
+  |*  [k=mold v=mold]
+  =>
+  ::  molds and mold builders
+  ::
+  |%
+  +$  elem  [=k p=@ =v]                                 ::  pri element
+  ::
+  +$  lnode  [n=elem l=ltree m=k r=ltree]               ::  loser tree node
```

Comment:
what is the difference between `k.n` and `m` here?

Reply:
`k.n` is the actual key corresponding to a `[priority value]` pair, whereas `m` is a 'split key' used to enforce so-called semi-heap & search tree structural properties.  You can see comparisons to `m` being used to determine balancing in `+put` and `+del`, for example.